### DISCUSSION TITLE
Curve Number Generator Spontaneously Stopped Functioning

### DISCUSSION THREAD

**Discussion by Brian-NEI** [ar-siddiqui/curve_number_generator#55] [Q&A] [answered]
One of our engineers who uses this software is getting errors from the curve number generator and I'm not sure how to resolve it for them.  The module went from reliably working to not operational without any obvious change.  I have fully reinstalled QGIS on their computer without improvement, so it may be something in the user profile that needs to be corrected.  **Do you have an idea what needs to be corrected to get their program working again?**

Note: We are using QGIS 3.28.11 for compatibility with a particular plug-in.
Note: A different user in our company experienced this issue before for a brief time.  It both spontaneously appeared and seemed to resolve itself after a couple days.  We can't count on that happening again.

Per our engineer, "I am attempting to generate the following layers: soils, land cover, impervious surface, and curve number using the curve number generator tool. When I do this, the tool runs at the beginning and then gives the error message / log below. Normally, the layers generate and I am able to view them in QGIS."

****

Log: QGIS version: 3.28.11-Firenze QGIS code revision: b02458a196 Qt version: 5.15.3 Python version: 3.9.5 GDAL version: 3.7.2 GEOS version: 3.12.0-CAPI-1.18.0 PROJ version: Rel. 9.3.0, September 1st, 2023 PDAL version: 2.5.5 (git-version: 24f78d) Algorithm started at: 2024-03-14T08:49:15 Algorithm 'Curve Number Generator (CONUS) (NLCD & SSURGO)' starting… Input parameters: { 'CnLookup' : None, 'CurveNumber' : 'TEMPORARY_OUTPUT', 'DrainedSoils' : False, 'NLCDImpervious' : 'TEMPORARY_OUTPUT', 'NLCDLandCover' : 'TEMPORARY_OUTPUT', 'Soils' : 'TEMPORARY_OUTPUT', 'aoi' : 'M:/Project/<projectnameredcated>/HydrologicStudyBuffer.gpkg' } Results: {'OUTPUT': 'Reprojected_71d1662b_93cb_4366_a131_b8471c8e7287'} Area Boundary layer extent area is 19,663.9762 acres Successfully downloaded https://www.mrlc.gov/geoserver/ows?version=1.1.0&SERVICE=WCS&VERSION=1.0.0&REQUEST=GetCoverage&FORMAT=GeoTIFF&COVERAGE=mrlc_download:NLCD_2021_Impervious_L48&CRS=EPSG:5070&WIDTH=326&HEIGHT=275&BBOX=908370.1039206127,1945791.3907935026,918146.5592071054,1954041.3413574635& Results: {'OUTPUT': 'C:/Users/<username>/AppData/Local/Temp/processing_Hdcrgq/ef1828aaed7c48ea94d89a85c19cfb66/OUTPUT.file'} GDAL command: gdalwarp -overwrite -t_srs ESRI:103041 -r near -of netCDF C:/Users/<username>/AppData/Local/Temp/processing_Hdcrgq/ef1828aaed7c48ea94d89a85c19cfb66/OUTPUT.file C:/Users/<username>/AppData/Local/Temp/processing_Hdcrgq/5bb6b6b0f651480dada9d9b6fde560eb/OUTPUT.nc GDAL command output: Copying color table from C:/Users/<username>/AppData/Local/Temp/processing_Hdcrgq/ef1828aaed7c48ea94d89a85c19cfb66/OUTPUT.file to new file. Creating output file that is 357P x 306L. Processing C:/Users/<username>/AppData/Local/Temp/processing_Hdcrgq/ef1828aaed7c48ea94d89a85c19cfb66/OUTPUT.file [1/1] : 0Using internal nodata values (e.g. 0) for image C:/Users/<username>/AppData/Local/Temp/processing_Hdcrgq/ef1828aaed7c48ea94d89a85c19cfb66/OUTPUT.file. Copying nodata values from source C:/Users/<username>/AppData/Local/Temp/processing_Hdcrgq/ef1828aaed7c48ea94d89a85c19cfb66/OUTPUT.file to destination C:/Users/<username>/AppData/Local/Temp/processing_Hdcrgq/5bb6b6b0f651480dada9d9b6fde560eb/OUTPUT.nc. ...10...20...30...40...50...60...70...80...90...100 - done. Process completed successfully Results: {'OUTPUT': 'C:/Users/<username>/AppData/Local/Temp/processing_Hdcrgq/5bb6b6b0f651480dada9d9b6fde560eb/OUTPUT.nc'} Successfully downloaded https://www.mrlc.gov/geoserver/ows?version=1.1.0&SERVICE=WCS&VERSION=1.0.0&REQUEST=GetCoverage&FORMAT=GeoTIFF&COVERAGE=mrlc_download:NLCD_2021_Land_Cover_L48&CRS=EPSG:5070&WIDTH=326&HEIGHT=275&BBOX=908370.1039206127,1945791.3907935026,918146.5592071054,1954041.3413574635& Results: {'OUTPUT': 'C:/Users/<username>/AppData/Local/Temp/processing_Hdcrgq/5d9da4ba91f7498aac60257f85d33c76/OUTPUT.file'} GDAL command: gdalwarp -overwrite -t_srs ESRI:103041 -r near -of netCDF C:/Users/<username>/AppData/Local/Temp/processing_Hdcrgq/5d9da4ba91f7498aac60257f85d33c76/OUTPUT.file C:/Users/<username>/AppData/Local/Temp/processing_Hdcrgq/3ef0de185f5f4516b7c8e8f4829e943f/OUTPUT.nc GDAL command output: Copying color table from C:/Users/<username>/AppData/Local/Temp/processing_Hdcrgq/5d9da4ba91f7498aac60257f85d33c76/OUTPUT.file to new file. Creating output file that is 357P x 306L. Processing C:/Users/<username>/AppData/Local/Temp/processing_Hdcrgq/5d9da4ba91f7498aac60257f85d33c76/OUTPUT.file [1/1] : 0Using internal nodata values (e.g. 0) for image C:/Users/<username>/AppData/Local/Temp/processing_Hdcrgq/5d9da4ba91f7498aac60257f85d33c76/OUTPUT.file. Copying nodata values from source C:/Users/<username>/AppData/Local/Temp/processing_Hdcrgq/5d9da4ba91f7498aac60257f85d33c76/OUTPUT.file to destination C:/Users/<username>/AppData/Local/Temp/processing_Hdcrgq/3ef0de185f5f4516b7c8e8f4829e943f/OUTPUT.nc. ...10...20...30...40...50...60...70...80...90...100 - done. Process completed successfully Results: {'OUTPUT': 'C:/Users/<username>/AppData/Local/Temp/processing_Hdcrgq/3ef0de185f5f4516b7c8e8f4829e943f/OUTPUT.nc'} Results: {'OUTPUT': 'Reprojected_cb8d5353_8a18_445d_af66_5b058b2dafe0'} Creating POST request... Results: {'OUTPUT': 'Fixed_geometries_727f3815_10a9_488c_8702_425308fd686c'} No spatial index exists for input layer, performance will be severely degraded Results: {'OUTPUT': 'Clipped_cfda1c09_d4c0_4862_97cd_6a3ea917d80f'} Results: {'OUTPUT': 'Reprojected_481f6b4c_2a15_4780_b2ac_5b8e01b2d063'} Results: {'OUTPUT': 'Fixed_geometries_0971c7f7_fcb7_4881_b489_45115be5f056'} GDAL command: gdal_polygonize.bat C:/Users/<username>/AppData/Local/Temp/processing_Hdcrgq/3ef0de185f5f4516b7c8e8f4829e943f/OUTPUT.nc -b 1 -f "Interlis 2" C:/Users/<username>/AppData/Local/Temp/processing_Hdcrgq/19e96d48d4fe4873beb284c24d3d6b30/OUTPUT.xtf OUTPUT land_cover GDAL command output: Creating output C:/Users/<username>/AppData/Local/Temp/processing_Hdcrgq/19e96d48d4fe4873beb284c24d3d6b30/OUTPUT.xtf of format Interlis 2. Traceback (most recent call last): File "C:\PROGRA~1\QGIS32~1.11\apps\Python39\lib\site-packages\osgeo_utils\auxiliary\gdal_argparse.py", line 175, in main self.doit(**kwargs) File "C:\PROGRA~1\QGIS32~1.11\apps\Python39\lib\site-packages\osgeo_utils\gdal_polygonize.py", line 314, in doit return gdal_polygonize(**kwargs) File "C:\PROGRA~1\QGIS32~1.11\apps\Python39\lib\site-packages\osgeo_utils\gdal_polygonize.py", line 132, in gdal_polygonize dst_layer = dst_ds.CreateLayer( AttributeError: 'NoneType' object has no attribute 'CreateLayer' Process returned error code 1 Results: {'OUTPUT': 'C:/Users/<username>/AppData/Local/Temp/processing_Hdcrgq/19e96d48d4fe4873beb284c24d3d6b30/OUTPUT.xtf'} Unable to execute algorithm Could not load source layer for INPUT: C:/Users/<username>/AppData/Local/Temp/processing_Hdcrgq/19e96d48d4fe4873beb284c24d3d6b30/OUTPUT.xtf not found Unable to execute algorithm Could not load source layer for INPUT: C:/Users/<username>/AppData/Local/Temp/processing_Hdcrgq/19e96d48d4fe4873beb284c24d3d6b30/OUTPUT.xtf not found Execution failed after 30.09 seconds Loading resulting layers Algorithm 'Curve Number Generator (CONUS) (NLCD & SSURGO)' finished

**Comment by ar-siddiqui** [comment, marked as answer]
Hi Brian, it looks like the user might have recently worked with NetCDF and some .xtf format which might have altered their default formats settings etc. Can you ask them to create a new profile in QGIS and try again?

**Reply by Brian-NEI** [reply]
That worked as a general solution, thank you.  Do you have an idea of a more precise setting which might be the root of the issue?  Our engineer didn't recognize either NetCDF or .xtf as something they worked on.

**Reply by ar-siddiqui** [reply]
This is probably a QGIS bug or some other plugin might be changing the settings for the default format. https://github.com/qgis/QGIS/issues/54952
